app/services/katello/pulp3/repository.rb

```diff
@@ -166,13 +166,18 @@ def compute_remote_options(computed_options = remote_options)
         computed_options.except(:name, :client_key)
       end
 
-      def create
-        unless repository_reference
+      def create(force = false)
+        if force || !repository_reference
           response = api.repositories_api.create(create_options)
-          RepositoryReference.create!(
-           root_repository_id: repo.root_id,
-           content_view_id: repo.content_view.id,
-           repository_href: response.pulp_href)
+          if force
+            RepositoryReference.where(
+              root_repository_id: repo.root_id,
+              content_view_id: repo.content_view.id).destroy_all
+          end
```

Comment:
would it make sense to just destroy regardless of the force option?   I think it'd simplify the code and prevent possible error conditions

Reply:
Yeah I think so, that'll simplify things.